### ISSUE TITLE
Doesnt execute on Mobile Safari iOS 11.4.1

### ISSUE THREAD

**Issue by BP9906** [Valve/fingerprintjs2#342]
I'm using the latest master branch and when 1.8.0 tries to get executed,  Mobile Safari loads the JS but nothing comes out.  If I revert back to a version from last year (1.5.1), it works just fine.

**Comment by jonashaag** [comment]
Please paste console output or other debug info. The info you provided is not much to go on

**Comment by BP9906** [comment]
I understand, but I literally get nothing using Safari Developer (Web Inspector). 
My js is written to console.log(result) and console.log(components) but I get nothing. Safari on Mac works fine and I get console data, but not from iOS Safari.  I can see in Web Inspector Network tab that the JS is loading just fine and the HTML for the test page loads fine too.  Web Inspector leads me to believe that when it attempts to execute the fingerprint2.js that is stops and terminates it, thus get get nothing.

**Comment by jonashaag** [comment]
Please provide a minimal reproducing example e.g. Jsfiddle

**Comment by BP9906** [comment]
Your example > https://jsfiddle.net/L2gLq4rg/  
This does not render any information in the white box on iPhone Mobile Safari.

**Comment by jonashaag** [comment]
Works for me. You have an ad blocker on iOS?

**Comment by BP9906** [comment]
Nope. I know that many users have the same issue because I’ve used it on a mobile site and we quickly got complaints only from iPhone users. 

I tried private browsing too. Same behavior. 
Running  iPhone 8 simulator on Mac OS  using the mobile safari gives me the same behavior.

**Comment by jonashaag** [comment]
Can you try to find the commit that caused the problem? It's 96 commits between 1.5.1 and master, so if you use binary search that's a maximum of 7 commits to check. Be sure to use the non-minified version (`fingerprint.js`)

**Comment by BP9906** [comment]
This commit, it works.  Narrowing.

commit de23695d09b5faa3d661c47ad0acc45040036b25
Author: Jonas Haag <jonas@lophus.org>
Date:   Mon Apr 9 18:03:31 2018 +0200

    Version 1.8.0

**Comment by jonashaag** [comment]
Hm have you tried the latest master at all? We've added error handling to audio fingerprinting a few commits after caf14da. Also, does it work if you `excludeAudioFP`?

To see developer console errors you can also attach macOS Safari developer tools with a USB cable (enable debugging somewhere in the iPhone Safari/developer settings)

**Comment by BP9906** [comment]
Yup, been using the MacOS Safari Developer USB cable method for a while now for testing.  I tried 
`var options = {extendedJsFonts: true, excludeAudioFP: true};`

Then attempting using the latest master and still get nothing. There are 0 console errors.  For sanity test, I added "console.log(variable)" in several places.  The only thing that I can get logged out is the object variable.
`
var fp = new Fingerprint2(preprocess,options);
console.log(fp);

fp.get(function(result, components){
`
The fp.get doesnt execute, so I dont get anything inside that function.
The console.log(fp) does dump the object.

**Comment by jonashaag** [comment]
FYI the exclude option name has changed from `excludeAudioFP` to `excludeAudio` in master.

**Comment by BP9906** [comment]
I tried excludeAudio and that didnt help either.

**Comment by yesvods** [comment]
same issue with ios11.4.1

**Comment by yesvods** [comment]
I found that `context.oncomplete` will not trigger and execute in IOS 11.4.1. So that the get method will broken.

**Comment by BP9906** [comment]
Master looks great; thanks for the fix!
